src/components/connections/Connection.ts

```diff
@@ -42,63 +44,80 @@ export class InfluxDBTreeDataProvider
     this._onDidChangeTreeData.fire(element);
   }
 
-  public async addConnection() {
-    const name = await vscode.window.showInputBox({
-      prompt: "Give a name of the influxdb connection",
-      placeHolder: "name",
-      ignoreFocusOut: true
-    });
-    if (!name) {
-      return;
-    }
-
-    const hostNport = await vscode.window.showInputBox({
-      prompt: "The hostname and port of the influxdb",
-      placeHolder: "http://localhost:9999",
-      ignoreFocusOut: true
-    });
-    if (!hostNport) {
-      return;
-    }
-
-    const token = await vscode.window.showInputBox({
-      prompt: "token of the influxdb",
-      placeHolder: "token",
-      ignoreFocusOut: true
-    });
-    if (!token) {
-      return;
-    }
-
-    const org = await vscode.window.showInputBox({
-      prompt: "name of the influxdb org",
-      placeHolder: "org name",
-      ignoreFocusOut: true
-    });
-    if (!org) {
-      return;
-    }
-
-    let connections = this.context.globalState.get<{
-      [key: string]: InfluxDBConnection;
-    }>(InfluxDBConectionsKey);
-
-    if (!connections) {
-      connections = {};
+  public async addConnection(context: ExtensionContext) {
+    let defaultURL = "";
+    let defaultURLConfig = vscode.workspace.getConfiguration("vsflux");
+    if (defaultURLConfig !== undefined) {
+      if (defaultURLConfig.get<string>("defaultInfluxDBURL") !== undefined) {
+        defaultURL = defaultURLConfig.get<string>(
+          "defaultInfluxDBURL"
+        ) as string;
```

Comment:
with the generics on `get` is this casting required?

Reply:
yeah, the get<string> returns string|undefined, so still require casting